### PR TITLE
Adding NET 4.8 WinForms Accessibility changes compatibility document.

### PR DESCRIPTION
**Pull request by M-Lipin**
Adding **winforms-accessibility-changes-48.md** document which contains the description of accessibility changes introduced in .NET 4.8:
* DataGridView control Accessibility changes;
* Improvements for screen readers (Narrator);
* CheckedListBox control Accessibility changes;
* ComboBox control Accessibility changes;
* LinkLabel control Accessibility changes;
* ProgressBar control Accessibility changes;
* PropertyGrid control Accessibility changes;
* Tool Strip control Accessibility changes;
* Visual cues improvements;

**Note:** the document is not final version and should be updated with the changes related to Keyboard ToolTip, High contrast changes and other accessibility improvements introduced by WinForms team for .NET 4.8.

//cc @PriyaPurkayastha

### CODE REVIEW

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- Changes to interaction with Narrator.
+- Changes in Accessible hierarchy. (improving navigation through UI Automation tree)
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+__How to opt in or out of these changes__
+  
+In order for the application to benefit from these changes, it must run on the .NET Framework 4.8. The application can benefit from these changes in either of the following ways:
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
In order for the application to benefit from these changes, it must run on the .NET Framework 4.8. The application can opt-in into these changes in either of the following ways:
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved DataGridView Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications and other UI Automation features.
+- The UI Automation element which corresponds to the <xref:System.Windows.Forms.DataGridViewComboBoxEditingControl> or <xref:System.Windows.Forms.DataGridViewTextBoxEditingControl> is now a child of corresponding editing cell.
+
+__Improved Link Label Accessibility support__
+- Improved <xref:System.Windows.Forms.LinkLabel> control accessibility: Narrator announces the disabled state for the link if corresponding <xref:System.Windows.Forms.LinkLabel> control is disabled.
+
+__Improved Progress Bar Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ProgressBar> control with ability to use UI Automation notifications and other UI Automation features. Developers are now able to use UI Automation notifications which Narrator can announce to indicate the progress.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]: @M-Lipin  - is there a link to UIA notification you could add here?

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+- Improved Narrator support for <xref:System.Windows.Forms.CheckedListBox> control. At this moment when navigating to the <xref:System.Windows.Forms.CheckedListBox> control using keyboard, Narrator focuses the <xref:System.Windows.Forms.CheckedListBox> item and announces it.
+
+__Improved ComboBox Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ComboBox> control with ability to use UI Automation notifications and other UI Automation features.
+
+__Improved DataGridView Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications and other UI Automation features.
+- The UI Automation element which corresponds to the <xref:System.Windows.Forms.DataGridViewComboBoxEditingControl> or <xref:System.Windows.Forms.DataGridViewTextBoxEditingControl> is now a child of corresponding editing cell.
+
+__Improved Link Label Accessibility support__
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
__Improved LinkLabel Accessibility support__
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- Improved <xref:System.Windows.Forms.LinkLabel> control accessibility: Narrator announces the disabled state for the link if corresponding <xref:System.Windows.Forms.LinkLabel> control is disabled.
+
+__Improved Progress Bar Accessibility support__
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
__Improved ProgressBar Accessibility support__
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+- Narrator now announces the sort direction of the <xref:System.Windows.Forms.DataGridViewColumn> when announcing an accessible name of a <xref:System.Windows.Forms.DataGridViewCell>. 
+
+__Improved Checked ListBox Accessibility support__
+- Improved Narrator support for <xref:System.Windows.Forms.CheckedListBox> control. At this moment when navigating to the <xref:System.Windows.Forms.CheckedListBox> control using keyboard, Narrator focuses the <xref:System.Windows.Forms.CheckedListBox> item and announces it.
+
+__Improved ComboBox Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ComboBox> control with ability to use UI Automation notifications and other UI Automation features.
+
+__Improved DataGridView Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications and other UI Automation features.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications, other UI Automation features, and keyboard navigation of the control.
```
Though I'm not 100% sure, please verify, I think we added f3 for sort order access

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+- The UI Automation element which corresponds to the <xref:System.Windows.Forms.DataGridViewComboBoxEditingControl> or <xref:System.Windows.Forms.DataGridViewTextBoxEditingControl> is now a child of corresponding editing cell.
+
+__Improved Link Label Accessibility support__
+- Improved <xref:System.Windows.Forms.LinkLabel> control accessibility: Narrator announces the disabled state for the link if corresponding <xref:System.Windows.Forms.LinkLabel> control is disabled.
+
+__Improved Progress Bar Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ProgressBar> control with ability to use UI Automation notifications and other UI Automation features. Developers are now able to use UI Automation notifications which Narrator can announce to indicate the progress.
+
+__Improved Property Grid Accessibility support__
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
__Improved PropertyGrid Accessibility support__
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+- Enabled UI Automation support for <xref:System.Windows.Forms.PropertyGrid> control with ability to use UI Automation notifications and other UI Automation features.
+- The UI Automation element which corresponds to the currently editing property is now a child of corresponding property item UI Automation element.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- The UI Automation element which corresponds to the currently edited property is now a child of the corresponding property item UI Automation element.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+- The UI Automation property item element is now a child of corresponding category if the parent <xref:System.Windows.Forms.PropertyGrid> control is set to Category view.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- The UI Automation property item element is now a child of the corresponding category element if the parent <xref:System.Windows.Forms.PropertyGrid> control is set to category view.
```
I'm not sure about capitalization of the word `category`.

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved Tool Strip support__
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
__Improved ToolStrip support__
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+- Enabled UI Automation support for <xref:System.Windows.Forms.ToolStrip> control with ability to use UI Automation notifications and other UI Automation features.
+- Improved navigation through <xref:System.Windows.Forms.ToolStrip> items. 
+- In items mode Narrator focus does not disappear and does not go to hidden items.
+
+__Improved Visual cues__
+- An empty <xref:System.Windows.Forms.CheckedListBox> control now displays a focus indicator when receives focus.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- An empty <xref:System.Windows.Forms.CheckedListBox> control now displays a focus indicator when it receives focus.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+Major
+
+### Version Introduced
+4.8
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+Windows Forms Framework is improving how it works with accessibility technologies to better support Windows Forms customers. These include the following changes:
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
The Windows Forms Framework is continuing to improve how it works with accessibility technologies to better support Windows Forms customers. These include the following changes:
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+- Changes to improve display during High Contrast mode.
+- Changes to interaction with Narrator.
+- Changes in Accessible hierarchy. (improving navigation through UI Automation tree)
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Changes in the Accessible hierarchy (improving navigation through the UI Automation tree).
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+__How to opt in or out of these changes__
+  
+In order for the application to benefit from these changes, it must run on the .NET Framework 4.8. The application can benefit from these changes in either of the following ways:
+- It is recompiled to target the .NET Framework 4.8. These accessibility changes are enabled by default on Windows Forms applications that target the .NET Framework 4.8.
+- It targets the .NET Framework 4.7.2 or earlier version and opts out of the legacy accessibility behaviors by adding the following [AppContext Switch](https://docs.microsoft.com/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element) to the `<runtime>` section of the app config file and setting it to `false`, as the following example shows.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- It targets the .NET Framework 4.7.2 or earlier version and opts out of the legacy accessibility behaviors by adding the following [AppContext switch](https://docs.microsoft.com/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element) to the `<runtime>` section of the app config file and setting it to `false`, as the following example shows.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+Applications that target the .NET Framework 4.8 and want to preserve the legacy accessibility behavior can opt in to the use of legacy accessibility features by explicitly setting this AppContext switch to `true`.
+
+__Use of OS-defined colors in High Contrast themes__
+// To be filled.
+
+__Improved Narrator support__
+- Narrator now announces the sort direction of the <xref:System.Windows.Forms.DataGridViewColumn> when announcing an accessible name of a <xref:System.Windows.Forms.DataGridViewCell>. 
+
+__Improved Checked ListBox Accessibility support__
+- Improved Narrator support for <xref:System.Windows.Forms.CheckedListBox> control. At this moment when navigating to the <xref:System.Windows.Forms.CheckedListBox> control using keyboard, Narrator focuses the <xref:System.Windows.Forms.CheckedListBox> item and announces it.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Improved Narrator support for the <xref:System.Windows.Forms.CheckedListBox> control. When navigating to the <xref:System.Windows.Forms.CheckedListBox> control using the keyboard, Narrator focuses the <xref:System.Windows.Forms.CheckedListBox> item and announces it.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved ComboBox Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ComboBox> control with ability to use UI Automation notifications and other UI Automation features.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Enabled UI Automation support for the <xref:System.Windows.Forms.ComboBox> control, with the ability to use UI Automation notifications and other UI Automation features.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved DataGridView Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications and other UI Automation features.
+- The UI Automation element which corresponds to the <xref:System.Windows.Forms.DataGridViewComboBoxEditingControl> or <xref:System.Windows.Forms.DataGridViewTextBoxEditingControl> is now a child of corresponding editing cell.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- The UI Automation element which corresponds to the <xref:System.Windows.Forms.DataGridViewComboBoxEditingControl> or <xref:System.Windows.Forms.DataGridViewTextBoxEditingControl> is now a child of the corresponding editing cell.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved Link Label Accessibility support__
+- Improved <xref:System.Windows.Forms.LinkLabel> control accessibility: Narrator announces the disabled state for the link if corresponding <xref:System.Windows.Forms.LinkLabel> control is disabled.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Improved <xref:System.Windows.Forms.LinkLabel> control accessibility: Narrator announces the disabled state for the link if the corresponding <xref:System.Windows.Forms.LinkLabel> control is disabled.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 2 above)
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Enabled UI Automation support for the <xref:System.Windows.Forms.ProgressBar> control with the ability to use UI Automation notifications and other UI Automation features. Developers are now able to use UI Automation notifications which Narrator can announce to indicate progress.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+__Improved Progress Bar Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ProgressBar> control with ability to use UI Automation notifications and other UI Automation features. Developers are now able to use UI Automation notifications which Narrator can announce to indicate the progress.
+
+__Improved Property Grid Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.PropertyGrid> control with ability to use UI Automation notifications and other UI Automation features.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Enabled UI Automation support for the <xref:System.Windows.Forms.PropertyGrid> control, with the ability to use UI Automation notifications and other UI Automation features.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+- The UI Automation element which corresponds to the currently editing property is now a child of corresponding property item UI Automation element.
+- The UI Automation property item element is now a child of corresponding category if the parent <xref:System.Windows.Forms.PropertyGrid> control is set to Category view.
+
+__Improved Tool Strip support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ToolStrip> control with ability to use UI Automation notifications and other UI Automation features.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- Enabled UI Automation support for the <xref:System.Windows.Forms.ToolStrip> control, with the ability to use UI Automation notifications and other UI Automation features.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+- Improved navigation through <xref:System.Windows.Forms.ToolStrip> items. 
+- In items mode Narrator focus does not disappear and does not go to hidden items.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
- In items mode, Narrator focus does not disappear and does not go to hidden items.
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 2 above)
Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: Thanks for the comment!
I've added the link in 15c3da6c5585bdd314b3b75a285626a016c5d011.

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 3 above)
Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: Thanks!

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 5 above)
Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: I looked at this and found out that F3-sorting was added in 4.7.2.
.

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 7 above)
Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: Thanks!

File: Documentation/compatibility/winforms-accessibility-changes-48.md (hunk 8 above)
Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: Thanks. I agree, category should not have 'c' capitalized.

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+533226
+657355
+661319
+642548
+537224
+549558
+581351
+560840
+497307
+
```
Comment (bryukov) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
548792
686499

```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+- Enabled UI Automation support for the <xref:System.Windows.Forms.ToolStrip> control, with the ability to use UI Automation notifications and other UI Automation features.
+- Improved navigation through <xref:System.Windows.Forms.ToolStrip> items. 
+- In items mode, Narrator focus does not disappear and does not go to hidden items.
+
+__Improved Visual cues__
+- An empty <xref:System.Windows.Forms.CheckedListBox> control now displays a focus indicator when it receives focus.
+
+For an overview of UI automation, see the [UI Automation Overview](https://docs.microsoft.com/dotnet/framework/ui-automation/ui-automation-overview).</p>
+
+
```
Comment (bryukov) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
__Invoking controls' ToolTips with a keyboard__
 Control tooltip can now be invoked by focusing the control with keyboard. 
 This feature needs to be enabled explicitly for the application (see section __"How to opt in or out of these changes"__)
 
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+      </startup>
+      <runtime>
+        <!-- AppContextSwitchOverrides value attribute is in the form of 'key1=true|false;key2=true|false  -->
+        <AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false" />
+      </runtime>
+    </configuration>
+```    
+Note that to opt in to the accessibility features added in .NET Framework 4.8, you must also opt in to accessibility features of .NET Framework 4.7.1 and 4.7.2 as well.
+Applications that target the .NET Framework 4.8 and want to preserve the legacy accessibility behavior can opt in to the use of legacy accessibility features by explicitly setting this AppContext switch to `true`.
+
```
Comment (bryukov) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
Enabling the keyboard ToolTip invocation support requires adding the `Switch.System.Windows.Forms.UseLegacyToolTipDisplay=false` line to the AppContextSwitchOverrides value:
`<AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false" />`

Note that enabling this feature requires opting in to the aforementioned accessibility features of .NET Framework 4.7.1 - 4.8.

```

**Comment by M-Lipin** [comment]
Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 544592: [Accessibility][Keyboard Navigation - BizTalk Server - Application - Send Port Gorups] Navigation is not cosistent when navigating in forward and backward direction on general window of Send port groups properties.
Change description: Improved ToolStrip keyboard navigation, made the ToolStrip forward and backward navigation consistent.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 544471: [Accessibility] DataGridView is reporting cells as always editable via Narrator.
Change description: made announcement for read-only DataGridView TextBox cell consistent: Narrator announces the cell as read-only.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 549288: Accessibility: WinForms ListView and DataGridView do not expose the current sort order through UIA.
Change description: The accessibility of DataGridView and ListView has been improved to make sort direction available via UIA: added exposing sort order and sort column via ItemStatus property and column name.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 599936: Accessibility:[Narrator]Narrator cannot annouce the ReadOnly Status for DataGridView TextBox column with setting Readonly as true.
Change description: The announcement of DataGridView Read-only TextBox cells is made accessible and clear: editable TextBoxes are announced as 'editable' and read-only TextBoxes are announced as 'read-only'.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 508369: Accessibility: [Narrator] Narrator is not announcing the updated value of ComboBox in PropertyGrid.
Change description: Added screen reader announcement for PropertyGrid's ComboBox value changes. Added ComboBox's text field update in response of ComboBox selection changes.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 615721: [Accessibility] The Invoke Pattern for DataGridViewCell: Image is not available.
Change description: Added support of Accessibility Invoke pattern to DataGridView Image cell with ability to invoke cell's default action. (in case Image cell is actually an image of a button)

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 574309: Accessibility: MAS40B: Inspect: Expand/Collapse pattern property is FALSE for 'Type' & ‘Scope’ comboBoxes in 'Project properties node-->Settings page'.
Change description: Made the DataGridView's Combobox expand/collapse state accessible for the customers which use assistive technologies.

**Comment by M-Lipin** [comment]
> Need to confirm that the following should (or should not) be added (not sure that these are in the email list but have 4.8 target):

Bug 564762: [Accessibility] The blue rectangle of selected ListView item should be keeping after tab or mouse to other control.
Change description: Improved ListView accessibility by enabling showing grey rectangle around the selected ListView item when parent ListView is out if focus.

**Comment by rpetrusha** [comment]
Is this ready to merge, @M-Lipin?

**Comment by M-Lipin** [comment]
> Is this ready to merge, @M-Lipin?

Hello @rpetrusha, I'll ask WinForms team for additional review, but basically it seems to me that this can be merged. Just need final confirmation.

**Comment by rpetrusha** [comment]
OK. Thanks, @M-Lipin. Let me know when it can be merged.

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+      <runtime>
+        <!-- AppContextSwitchOverrides value attribute is in the form of 'key1=true|false;key2=true|false  -->
+        <AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false" />
+      </runtime>
+    </configuration>
+```    
+Note that to opt in to the accessibility features added in .NET Framework 4.8, you must also opt in to accessibility features of .NET Framework 4.7.1 and 4.7.2 as well.
+Applications that target the .NET Framework 4.8 and want to preserve the legacy accessibility behavior can opt in to the use of legacy accessibility features by explicitly setting this AppContext switch to `true`.
+Enabling the keyboard ToolTip invocation support requires adding the `Switch.System.Windows.Forms.UseLegacyToolTipDisplay=false` line to the AppContextSwitchOverrides value:
+`<AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false" />`
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]:
```suggestion
`<AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false;Switch.System.Windows.Forms.UseLegacyToolTIpDisplay=false" />`
```

File: Documentation/compatibility/winforms-accessibility-changes-48.md [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+
+Note that enabling this feature requires opting in to the aforementioned accessibility features of .NET Framework 4.7.1 - 4.8.
```
Comment (Tanya-Solyanik) [inline Documentation/compatibility/winforms-accessibility-changes-48.md]: I would add that if any of the accessibility features are not opted in but the tooltip display feature is opted in, users would see a runtime exception on the first access to these features, most likely on application start up. Please specify exception type here.

Comment (M-Lipin) [reply Documentation/compatibility/winforms-accessibility-changes-48.md]: Thank you, I've added a note about enabling keyboard tooltip feature as you suggested. Thanks!

**Pull request merged**